cmd/root.go

```diff
@@ -465,18 +452,48 @@ func runSignalWrapper(cmd *Command) error {
 		cmd.logger.Errorf("The proxy has encountered a terminal error: %v", err)
 		return err
 	case p = <-startCh:
+		cmd.logger.Infof("The proxy has started successfully and is ready for new connections!")
 	}
-	cmd.logger.Infof("The proxy has started successfully and is ready for new connections!")
-	defer p.Close()
 	defer func() {
 		if cErr := p.Close(); cErr != nil {
 			cmd.logger.Errorf("error during shutdown: %v", cErr)
 		}
 	}()
 
-	go func() {
-		shutdownCh <- p.Serve(ctx)
-	}()
+	notify := func() {}
+	if cmd.healthCheck {
+		hc := healthcheck.NewCheck(p, cmd.logger)
+		mux.HandleFunc("/startup", hc.HandleStartup)
+		mux.HandleFunc("/readiness", hc.HandleReadiness)
+		mux.HandleFunc("/liveness", hc.HandleLiveness)
+		notify = hc.NotifyStarted
+	}
+
+	// Start the HTTP server is either Prometheus or the health check was
+	// enabled.
+	if cmd.prometheusNamespace != "" || cmd.healthCheck {
```

Comment:
Since we have to take different actions for prometheus vs health check, we do need two separate values. Nonetheless, looking again at prometheus, the namespace is optional, so I'll update the flag to be a boolean (and add a namespace flag separately), which cleans this up slightly.

Reply:
On second thought, I'll make this change as a separate PR to keep things focused here.

Reply:
So what I was trying to suggest was that the need for an http server is probably orthogonal to which specific httpPorts have been specified. 

So something like this:

```go
if cmd.prometheusNamespace != "" {
    cmd.needHttpServer = true
    // other health check stuff
}
if cmd.healthCheck {
    cmd.needHttpServer = true
    // other health check stuff
}

if cmd.needHttpServer {
     // start http server
}
```

(Consider this a nit)